src/main/java/org/truffleruby/interop/OutgoingForeignCallNode.java

```diff
@@ -69,45 +68,28 @@ public Object callCached(
             TruffleObject receiver,
             Object[] args,
             @Cached("args.length") int cachedArgsLength,
-            @Cached("createHelperNode(cachedArgsLength)") OutgoingNode outgoingNode,
-            @Cached("createToForeignNodes(cachedArgsLength)") RubyToForeignNode[] toForeignNodes,
-            @Cached("create()") ForeignToRubyNode toRubyNode) {
-        Object[] foreignArgs = argsToForeign(frame, toForeignNodes, args);
-        return doCall(frame, receiver, outgoingNode, foreignArgs, toRubyNode);
+            @Cached("createHelperNode(cachedArgsLength)") OutgoingNode outgoingNode) {
+        return doCall(frame, receiver, outgoingNode, args);
```

Comment:
Here is the difference, argsToForeign is no longer called on the arguments for a Ruby to foreign call.
This is going to break code, e.g. Fiddle, NFI, Ruby FFI and likely other interop.

Reply:
Yes, that's what's documented isn't it?

Ruby strings are unboxable - other languages should be the ones to unbox them to Java strings if they want that. NFI isn't doing it at the moment so I've opened an issue. This is also documented, and it's in the contract for interop.

Reply:
Right, it's documented, but this is a breaking change from the previous behavior. It might be better to do it later and have this PR only do more conversions, consistently everywhere (well, except C-exts), not less.

Reply:
Since j.l.String is considered a primitive, I agree the unboxing should be transparent and should work, but it's still a big bet that other languages will do it consistently, when most basically only use j.l.String.